alias.c

```diff
@@ -681,17 +648,21 @@ bool mutt_addr_is_user(struct Address *addr)
   return false;
 }
 
-void mutt_alias_free(struct Alias **p)
+void mutt_alias_free(struct Alias *p)
 {
-  struct Alias *t = NULL;
+  if (!p)
+    return;
+
+  mutt_alias_delete_reverse(p);
+  FREE(&p->name);
+  mutt_addr_free(&p->addr);
+  FREE(&p);
+}
 
-  while (*p)
+void mutt_aliaslist_free(struct AliasList *a_list) {
+  struct Alias *a = NULL, *tmp = NULL;
+  TAILQ_FOREACH_SAFE(a, a_list, entries, tmp)
   {
-    t = *p;
-    *p = (*p)->next;
-    mutt_alias_delete_reverse(t);
-    FREE(&t->name);
-    mutt_addr_free(&t->addr);
-    FREE(&t);
+    mutt_alias_free(a);
```

Comment:
I think we need a `REMOVE` here, otherwise we'll end up with a list of dead pointers.

Reply:
Fixed.